src/Domain/CampaignRepository.php

```diff
@@ -98,6 +98,45 @@ public function findOneBySalesforceId(Salesforce18Id $salesforceId): ?Campaign
         return $this->findOneBy(['salesforceId' => $salesforceId->value]);
     }
 
+    /**
+     * @param string $metaCampaginSlug
+     * @return array{newFetchCount: int, updatedCount: int}
+ *@throws NotFoundException
+     *
+     * @throws CampaignNotReady
+     */
+    public function fetchAllForMetaCampaign(string $metaCampaginSlug): array
+    {
+        /** @var list<array{id: string}> $campaignList */
+        $campaignList = $this->client->findCampaignsForMetaCampaign($metaCampaginSlug, limit: 10_000);
+
+        $campaignIds = array_map(function (array $campaign) {
+            return Salesforce18Id::ofCampaign($campaign['id']);
+        }, $campaignList);
+
+        $newFetchCount = 0;
+        $updatedCount = 0;
+
+        $count = count($campaignIds);
+
+        $i = 0;
+        foreach ($campaignIds as $id) {
+            $i++;
+            $this->logger->info("Fetching campaign $i of $count\n");
+            $existingCampaignInDB = $this->findOneBySalesforceId($id);
+
+            if ($existingCampaignInDB) {
+                $this->updateFromSf($existingCampaignInDB, withCache: false, autoSave: true);
```

Comment:
Should we also add `$this->fundRepository->pullForCampaign($campaign);` around here as you suggested earlier? I think that would make the effect pretty much identical to what `UpdateCampaigns` does for already known ones.

Reply:
Yes I think so. I didn't want to make CampaignRepo depend on FundRepo so I've done it as a second phase after this loop completes in f80ca87efa0

Reply:
Nice, looks good!